### PR TITLE
Refactored function names in R/lineage.R

### PR DESCRIPTION
**Pull request by teddyCodex**
# Description

Refactored specified function names in R/lineage.R based on issue #7 

Original | Modified | User Facing
| :---         |     :---    |          :--- |
GCA2lin | GCA2Lineage |  ✅|
add_tax | addTaxID |  ✅|
prot2tax | proteinAcc2TaxID |  ✅|
DownloadAssemblySummary | downloadAssemblySummary |  ✅|
prot2tax_old* | proteinAcc2TaxID_old | did not exist in lineage.R |

Fixes issue #48

## What kind of change(s) are included?

- [ ] Feature (adds or updates new capabilities)
- [x] Bug fix (fixes an issue).
- [ ] Enhancement (adds functionality).
- [ ] Breaking change (these changes would cause existing functionality to not work as expected).

# Checklist

Please ensure that all boxes are checked before indicating that this pull request is ready for review.

- [x] I have read and followed the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [x] I have searched for existing content to ensure this is not a duplicate.
- [x] I have performed a self-review of these additions (including spelling, grammar, and related).
- [x] I have added comments to my code to help provide understanding.
- [x] I have added a test which covers the code changes found within this PR.
- [x] I have deleted all non-relevant text in this pull request template.
- [x] **Reviewer assignment**: Tag a relevant team member to review and approve the changes.

### CODE REVIEW

**Comment by teddyCodex** [comment]
@jananiravi @the-mayer file containing locations where the functions are called or referenced.

[function_occurrences #48.txt](https://github.com/user-attachments/files/17278640/function_occurrences.48.txt)

File: R/acc2lin.R [hunk 1]
```diff
@@ -73,7 +73,7 @@ addLineage <- function(df, acc_col = "AccNum", assembly_path,
 #'
 #' @param accessions Character vector of protein accessions
 #' @param assembly_path String of the path to the assembly_summary path
-#' This file can be generated using the "DownloadAssemblySummary()" function
+#' This file can be generated using the \link[MolEvolvR]{downloadAssemblySummary} function
```
Comment (the-mayer) [inline R/acc2lin.R:76]: Added ref to new MolEvolvR function

File: R/lineage.R [hunk 2]
```diff
@@ -353,25 +353,25 @@ IPG2Lineage <- function(accessions, ipg_file,
...
-        genbank_lins <- GCA2lin(gca_ipg_dt,
+        genbank_lins <- GCA2Lineage(gca_ipg_dt,
             assembly_path = genbank_assembly_path,
             lineagelookup_path
         )
     }
 
 
-    lins <- GCA2lin(prot_data = ipg_dt, assembly_path, lineagelookup_path)
+    lins <- GCA2Lineage(prot_data = ipg_dt, assembly_path, lineagelookup_path)
```
Comment (the-mayer) [inline R/lineage.R:374]: Went ahead and updated to the new function name, where appropriate

**Review by the-mayer — APPROVED** [review]
Looks good, merging shortly!

**Pull request merged**